### DISCUSSION TITLE
How to install a new plugin?

### DISCUSSION THREAD

**Discussion by hditano** [NvChad/NvChad#1485] [General]
Hello everyone.

I'm new to LUA and trying to understand the documentation.
This is what I have done so far, so mind about my rookiness.

copied examples/init.lua to /nvim/lua/custom/plugins & chadrc.lua to /nvim/lua/custom/ directory.

Then I took a random plugin so I could make it work to learn how to install external plugins.

**/nvim/lua/custom/plugins/init.lua**
```
-- example file i.e lua/custom/init.lua
-- load your options globals, autocmds here or anything .__.
-- you can even override default options here (core/options.lua)
--
return {
  ["wakatime/vim-wakatime"] = {
    config = function()
      require("vim-wakatime").setup()
  },
}
```

**/nvim/lua/custom/chadrc.lua**
```
-- Just an example, supposed to be placed in /lua/custom/

local M = {}


M.plugins = {
  user = require("custom.plugins")
}
-- make sure you maintain the structure of `core/default_config.lua` here,
-- example of changing theme:

M.ui = {
  theme = "gruvchad",
}

return M
```

After installing everything :PackerSync :PackerStatus, relaunched neovim and nothing happens.
Is there anything that I am missing?

thx!

**Comment by hditano** [comment]
So after playing around with it an hour or so.
I was able to install an external plugin perfectly.

This is what I did:

![image](https://user-images.githubusercontent.com/6069906/185419344-6fe7a367-fbcb-4a5b-a863-e75a88821ffe.png)

**Reply by siduck** [reply]
Idk if there's a wakatime ft and adding that line will mean that wakatime will load only on that filetype
, i think wakatime tracks time spent on all sorts of filetypes right? so remove that ft line and check PackerStatus

**Reply by hditano** [reply]
didnt knew what ft means... sadly if I remove that line...wakatime will get uninstalled.
Is there any way to specific to track certain file extensions... something like

`["wakatime/vim-wakatime"] = {ft = "*.js,*.jsx,*.ts,*.json,*.html}`

**Reply by siduck** [reply]
@hditano example : 

```
ft = { "javascript", "javascriptreact, "html" }
```

search for correct filetype names from this command `Telescope filetypes`
